### ISSUE TITLE
`# noqa: E702 # fmt: skip` still formats the line of code

### ISSUE THREAD

**Issue by ngoctnq** [astral-sh/ruff#11430]
Input code:
```python
print(); print() # noqa # fmt: skip
```

After formatting:
```python
print()
print()  # noqa # fmt: skip
```

`# fmt: on/off` works fine though, but I want to have just one line of short boilerplate code.

I'm using the Ruff VS Code extension, which is said to ship with `ruff==0.4.1`

**Comment by dhruvmanila** [comment]
I think it's because there are two statements on the same line and `fmt: skip` is being applied only for the last statement. This can be seen by adding a single-quoted string (https://play.ruff.rs/edf37109-eb34-4b5c-9867-dd5587bef085), notice that the formatter only changes the quote for the first print call and not the second one. I don't think `RUF028` will indicate here because the suppression comment is actually being used by the formatter.

I'm not exactly sure if this is the intended behavior, @konstin any thoughts here? @MichaReiser  will be able to provide more context here but he's currently on PTO until the end of next week. I can possibly look into it to confirm whether this is the intended behavior.

**Comment by MichaReiser** [comment]
Oh, that's interesting and not something I have thought of when implementing suppression comments. 

> # fmt: skip comments suppress formatting for a preceding statement, case header, decorator, function definition, or class definition:

Today's behavior matches our documentation. However, Black doesn't format simple statements that end with a `fmt: skip` comment. So it is at least a Black incompatibility (and possibly not flagged by the ruff lint rule).

I'm open to changing our implementation to support this use case if it doesn't add too much complexity. I otherwise recommend documenting the deviation because it's possible to suppress the example using `fmt: off` (although it is slightly more verbose).

```python
# fmt: off
print(); print() # noqa 
# fmt: on

print( )
```

**Comment by Gatsby-Lee** [comment]
> Oh, that's interesting and not something I have thought of when implementing suppression comments.
> 
> > `# fmt: skip` comments suppress formatting for a preceding statement, case header, decorator, function definition, or class definition:
> 
> Today's behavior matches our documentation. However, Black doesn't format simple statements that end with a `fmt: skip` comment. So it is at least a Black incompatibility (and possibly not flagged by the ruff lint rule).
> 
> I'm open to changing our implementation to support this use case if it doesn't add too much complexity. I otherwise recommend documenting the deviation because it's possible to suppress the example using `fmt: off` (although it is slightly more verbose).
> 
> ```python
> # fmt: off
> print(); print() # noqa 
> # fmt: on
> 
> print( )
> ```

yep. basically, I banged my head a few hours to figure this out.
And, I finally just changed to Black formatter again.

Black doesn't format the line which has more than one statement and tagged with "# fmt: skip"

**Comment by Gatsby-Lee** [comment]
@ngoctnq 

thank you for writing this issue.

**Comment by Peiffap** [comment]
Related: #11216.

**Comment by dylwil3** [comment]
Resolved by https://github.com/astral-sh/ruff/pull/22119
